agent/hcp/telemetry/otel_sink.go

```diff
@@ -0,0 +1,189 @@
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/consul/agent/hcp/client"
+	"github.com/hashicorp/go-hclog"
+
+	gometrics "github.com/armon/go-metrics"
+	"go.opentelemetry.io/otel/attribute"
+	otelmetric "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/instrument"
+	otelsdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+type OTELSinkOpts struct {
```

Comment:
Keeping this struct as we have one last follow up PR that adds default filters and labels to the sink (so we'll have 4 parameters)